MatrixKit/Models/Contact/MXKContactManager.m

```diff
@@ -1833,17 +1836,19 @@ - (BOOL)encryptAndSaveData:(NSData*)data toFile:(NSString*)fileName
 
 - (NSData*)encryptData:(NSData*)data error:(NSError**)error fileName:(NSString*)fileName
 {
-    if (self.encryptionDelegate)
+    @try
     {
-        if (![self.encryptionDelegate isReady])
+        MXAesKeyData *keyData = (MXAesKeyData *) [[MXKeyProvider sharedInstance] requestKeyForDataOfType:kMXKContactManagerDataType isMandatory:NO expectedKeyType:kAes];
```

Comment:
Can we use `isKindOfClass`?

Reply:
Actually because the expected key type is AES, the returned value will be necessary of type MXAesKeyData (otherwise an exception is raised).

I can add this check but it's unnecessary though

Reply:
I was thinking about compiler warning, like unsafe cast.
Perhaps the app build settings are less restrictive. So, I would prefer to fix it for the compiler, not the algo behind.

Reply:
Understood. Nevertheless, we have no warning as I explicitly cast the type of Key. Warnings are displayed only in case of implicit cast. Should I proceed ?